common-api/src/main/scala/pl/touk/nussknacker/engine/api/parameter/FragmentParameterValueInput.scala

```diff
@@ -5,10 +5,11 @@ import pl.touk.nussknacker.engine.api.CirceUtil._
 import pl.touk.nussknacker.engine.api.definition.FixedExpressionValue
 
 @ConfiguredJsonCodec
-sealed trait ValueInputWithFixedValues {
+sealed trait FragmentParameterValueInput {
   def allowOtherValue: Boolean
-  def fixedValuesList: List[FixedExpressionValue]
 }
 
 case class ValueInputWithFixedValuesProvided(fixedValuesList: List[FixedExpressionValue], allowOtherValue: Boolean)
-    extends ValueInputWithFixedValues
+    extends FragmentParameterValueInput
+
+case class ValueInputWithDictEditor(dictId: String, allowOtherValue: Boolean) extends FragmentParameterValueInput
```

Comment:
I don't think it's crazy, the use case is similar to `ValueInputWithFixedValuesProvided` with `allowOtherValue = true`, either use a value from the dict or switch to raw editor and input whatever

I did however forgot to pattern match it in one place, now I've fixed it and change a test to detect it

Reply:
By "crazy" I mean that following this approach we introduce the possibility that user instead of extending the existing dictionary or creating another one, can ad-hoc use some value that is outside the dictionary. Than some another user can be confused that some value is missing from the dictionary. When this another user discover that we left such a hole, he/she can be angry with us that we allow such a bad pattern. But it is rather a UX thing, we can follow this approach and when we occur such a situation, we will blame our UX designers ;-) With regards to the code, if `allowOtherValue` is the common thing for both cases `ValueInputWithDictEditor` and `ValueInputWithFixedValuesProvided` we should extract a case class with this field and keep only differences in the `FragmentParameterValueInput` - so basically DRY, composition over inheritance. Thanks to that you will avoid such mistakes as missing handling logic in pattern matching .

Reply:
regarding extracting `allowOtherValue` etc - I agree, but changing it now would also change `ParameterAdditionalUIConfig`, so If someone is already using it, it would impose a migration

it's ofc doable, but i'm not sure if the small refactor is worth the hassle

Reply:
(and the missed pattern match was on DualParameterEditor, not really related directly to this structure)

Reply:
> allowOtherValue etc - I agree, but changing it now would also change ParameterAdditionalUIConfig, so If 
> someone is already using it, it would impose a need for a migration
This API is for customizable usages, on this level of adoption of our application, we shouldn't care too much about the compatibility of this API.

I'm not pushing to change it. I am allergic to the overuse of the inheritancebut this feature is quite niche so I think that we can live with such a technical debt.